saleor/graphql/product/mutations/products.py

```diff
@@ -1175,10 +1184,30 @@ def clean_input(
         cleaned_input = super().clean_input(info, instance, data)
 
         if "cost_price" in cleaned_input:
-            cleaned_input["cost_price_amount"] = cleaned_input.pop("cost_price")
+            cost_price = cleaned_input.pop("cost_price")
+            if cost_price < 0:
+                raise ValidationError(
+                    {
+                        "costPrice": ValidationError(
+                            ("Product price cannot be lower than 0."),
+                            code=ProductErrorCode.INVALID,
+                        )
+                    }
+                )
+            cleaned_input["cost_price_amount"] = cost_price
 
         if "price_override" in cleaned_input:
-            cleaned_input["price_override_amount"] = cleaned_input.pop("price_override")
+            price_override = cleaned_input.pop("price_override")
+            if price_override < 0:
```

Comment:
`ProductVariantUpdate` inherit form this mutation. There is a change that `price_override` in `None`. Did you check it?